### PR TITLE
docs: refactor pipe example to use the HttpClient

### PR DESCRIPTION
**Pull request by danielsogl**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Actually the pipes example is using the deprecated HttpModule.

## What is the new behavior?
Refactored the example to use the new HttpClient

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

**Comment by trotyl** [comment]
Partially covered by #22727

**Comment by brandonroberts** [comment]
@danielsogl will you rebase on master and remove `(aio)` from the git commit message?

**Comment by danielsogl** [comment]
@brandonroberts Done!

**Comment by brandonroberts** [comment]
Update the commit message to start with `docs: ...`

**Comment by danielsogl** [comment]
@brandonroberts fixed

**Review by brandonroberts — DISMISSED** [review]

**Comment by brandonroberts** [comment]
@danielsogl can you rebase this branch on master? Some of these changes have already landed in master

**Comment by brandonroberts** [comment]
@danielsogl something went off track with your rebase. Will you rebase on master again?

File: aio/content/examples/pipes/src/app/app.module.ts [hunk 1]
```diff
@@ -1,26 +1,23 @@
 // #docregion
+import { HttpClientModule } from '@angular/common/http';
```
Comment (brandonroberts) [inline aio/content/examples/pipes/src/app/app.module.ts]: An `HttpClientModule` import already exists

File: aio/content/examples/pipes/src/app/app.module.ts [hunk 2]
```diff
@@ -1,26 +1,23 @@
...
 import { NgModule } from '@angular/core';
-import { BrowserModule } from '@angular/platform-browser';
```
Comment (brandonroberts) [inline aio/content/examples/pipes/src/app/app.module.ts]: This should not be removed

File: aio/content/examples/pipes/src/app/fetch-json.pipe.ts [hunk 3]
```diff
@@ -1,23 +1,24 @@
 // #docregion
+import { HttpClient } from '@angular/common/http';
```
Comment (brandonroberts) [inline aio/content/examples/pipes/src/app/fetch-json.pipe.ts]: The `from` statements should be lined up. This is intentional.

**Review by brandonroberts — CHANGES_REQUESTED** [review]
There are still some changes outside the `aio` folder that need to be reverted.

**Comment by danielsogl** [comment]
@brandonroberts rebased on master again

File: aio/content/examples/pipes/src/app/app.module.ts [hunk 4]
```diff
@@ -43,6 +38,6 @@ import { ExponentialStrengthPipe } from './exponential-strength.pipe';
     FetchJsonPipe,
     ExponentialStrengthPipe
   ],
-  bootstrap: [ AppComponent ]
+  bootstrap: [AppComponent]
 })
-export class AppModule { }
+export class AppModule {}
```
Comment (gkalpak) [inline aio/content/examples/pipes/src/app/app.module.ts]: Please don't remove the spaces (here and above) for consistency.
(I don't like them either, but it is preferrable to remain consistent across examples and most examples use them.)

File: aio/content/examples/pipes/src/app/fetch-json.pipe.ts [hunk 5]
```diff
@@ -1,23 +1,24 @@
...
   pure: false
 })
 // #enddocregion pipe-metadata
-export class FetchJsonPipe  implements PipeTransform {
+export class FetchJsonPipe implements PipeTransform {
   private cachedData: any = null;
   private cachedUrl = '';
 
-  constructor(private http: HttpClient) { }
+  constructor(private http: HttpClient) {}
```
Comment (gkalpak) [inline aio/content/examples/pipes/src/app/fetch-json.pipe.ts]: Keep the space.

File: aio/content/examples/pipes/src/app/fetch-json.pipe.ts [hunk 6]
```diff
@@ -1,23 +1,24 @@
...
 
   transform(url: string): any {
     if (url !== this.cachedUrl) {
       this.cachedData = null;
       this.cachedUrl = url;
-      this.http.get(url).subscribe( result => this.cachedData = result );
+      this.http.get(url).subscribe(result => (this.cachedData = result));
```
Comment (gkalpak) [inline aio/content/examples/pipes/src/app/fetch-json.pipe.ts]: Why the parens?

File: aio/content/guide/architecture-components.md [hunk 7]
```diff
@@ -40,7 +40,7 @@ Angular inserts an instance of the `HeroListComponent` view between those tags.
 
 * `templateUrl`: The module-relative address of this component's HTML template. Alternatively, you can provide the HTML template inline, as the value of the `template` property. This template defines the component's _host view_.
 
-* `providers`: An array of **dependency injection providers** for services that the component requires. In the example, this tells Angular how to provide the `HeroService` instance that the component's constructor uses to get the list of heroes to display.  
+* `providers`: An array of **dependency injection providers** for services that the component requires. In the example, this tells Angular how provide the `HeroService` instance that the component's constructor uses to get the list of heroes to display.
```
Comment (gkalpak) [inline aio/content/guide/architecture-components.md]: Revert.

File: modules/benchmarks/src/tree/render3/tree.ts [hunk 8]
```diff
@@ -44,8 +44,8 @@ export class TreeComponent {
         s(null, c0);
         { T(1); }
         e();
-        C(2);
         C(3);
+        C(4);
```
Comment (gkalpak) [inline modules/benchmarks/src/tree/render3/tree.ts]: These changes seem irrelevant :grin:
(Probably a rebase gone bad.)

**Review by gkalpak — CHANGES_REQUESTED** [review]

**Comment by danielsogl** [comment]
@gkalpak yes I failed with the rebase. Now it should be fine

**Review by gkalpak — APPROVED** [review]
Looks great :smiley: Thx! 👍 
(Leaving it to @brandonroberts to give the final LGTM - assuming CI is green 🤔)

**Review by brandonroberts — APPROVED** [review]

**Pull request closed without merging**